src/main/java/org/sefglobal/scholarx/service/ProfileService.java

```diff
@@ -65,10 +64,24 @@ private Profile buildProfile(LinkedInAuthUserInfo oAuth2UserInfo) {
         profile.setLinkedinUrl("https://www.linkedin.com/search/results/all/?keywords="+
                 oAuth2UserInfo.getFirstName()+"%20"+
                 oAuth2UserInfo.getLastName()+"&origin=TYPEAHEAD_ESCAPE_HATCH");
+        profile.setHasConfirmedUserDetails(false);
         Date now = Calendar.getInstance().getTime();
         profile.setCreatedAt(now);
         profile.setUpdatedAt(now);
         return profile;
     }
 
+    private Profile updateUserDetails(long profileId, Profile profile)
+            throws ResourceNotFoundException, UnauthorizedException {
+        Optional <Profile> optionalUser = profileRepository.findByUid(profileId);
+        if (!optionalUser.isPresent()) {
+            String msg = "Error, Unable update details. " +
+                    "Profile with id: " + profileId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+        optionalUser.get().setEmail(profile.getEmail());
```

Comment:
We cannot replace the existing email field because the next time user gets logged in it will be replaced with the one from LinkedIn.  please create a new field and keep it there.

Reply:
Sure, Thanks for the review

Reply:
It doesn't replace the email again because kumuditha had removed that part from the user registration.(line 39 of this file)